### DISCUSSION TITLE
auth token on localstorage, how does security work? (next.js)

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#2301] [Q&A]
this project is incredible and won me over in such a crazy way that I can't get it out of my head for days 😅
but I'm not very good at the front end, and when I researched a lot about how pocketbase authentication works, I came across the token being stored in localstorage.
I took some basic examples of authentication that I found on youtube. but when logging in, I copied the localstorage content and put it in another browser and managed to access the account.

is it supposed to be like that? 👀

sorry if this question seems too dumb.

I'm making a whole system in nextjs with a backend node api (but its authentication is separate)
I want to use pocketbase to authenticate the user on the front end and be able to pull data from the db directly (only user data and some information that the backend node will insert by itself)

however this ease of copying the token and injecting it into another browser worries me.

I'm currently using supabase with all its pre-made components so I end up not worrying about this detail because it abstracts it for me, but I want to migrate to pocketbase for practicality and simplicity and for being self hosted and VERY LIGHT 🤩
(seriously, I was impressed by the low consumption of resources, I made some api floods to test pocketbase on a tiny vps and it didn't even show any bottleneck🤣)

I just wanted an explanation or something that can help me improve authentication security.

thank you for this amazing project 👍

**Comment by ganigeorgiev** [comment]
> I took some basic examples of authentication that I found on youtube. but when logging in, I copied the localstorage content and put it in another browser and managed to access the account.
> is it supposed to be like that? 

Yes. The same could happened if it was a session cookie and you've copied the cookie in another browser.

The only security concern that you should be aware with keeping the token in the `localStorage` is XSS, but modern frameworks and browsers help mitigating this with the built-in template engines and CSP policies.
You may also want to check https://github.com/pocketbase/js-sdk#security.
